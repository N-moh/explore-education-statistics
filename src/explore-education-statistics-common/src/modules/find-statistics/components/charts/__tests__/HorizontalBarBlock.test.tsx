import React from 'react';

import { render } from 'react-testing-library';

import HorzontalBarBlock from '../HorizontalBarBlock';

import testData from './__data__/testBlockData';

jest.mock('recharts/lib/util/LogUtils');

const props = testData.AbstractChartProps;
const { axes } = props;

describe('HorzontalBarBlock', () => {
  test('renders basic chart correctly', () => {
    const { container } = render(<HorzontalBarBlock {...props} width={900} />);

    expect(container).toMatchSnapshot();

    // axes
    expect(
      container.querySelector('.recharts-cartesian-axis.xAxis'),
    ).toBeInTheDocument();
    expect(
      container.querySelector('.recharts-cartesian-axis.yAxis'),
    ).toBeInTheDocument();

    // grid & grid lines
    expect(
      container.querySelector('.recharts-cartesian-grid'),
    ).toBeInTheDocument();
    expect(
      container.querySelector('.recharts-cartesian-grid-horizontal'),
    ).toBeInTheDocument();
    expect(
      container.querySelector('.recharts-cartesian-grid-vertical'),
    ).toBeInTheDocument();

    expect(
      container.querySelector('.recharts-default-legend'),
    ).toBeInTheDocument();

    // expect there to be rectangles for all 3 data sets across both years
    expect(
      Array.from(container.querySelectorAll('.recharts-rectangle')).length,
    ).toBe(6);
  });

  test('major axis can be hidden', () => {
    const { container } = render(
      <HorzontalBarBlock
        {...props}
        axes={{
          ...axes,
          major: {
            ...axes.major,
            visible: false,
          },
        }}
      />,
    );

    expect(
      container.querySelector('.recharts-cartesian-axis.xAxis'),
    ).not.toBeInTheDocument();
  });

  test('minor axis can be hidden', () => {
    const { container } = render(
      <HorzontalBarBlock
        {...props}
        axes={{
          ...axes,
          minor: {
            ...axes.minor,
            visible: false,
          },
        }}
      />,
    );

    expect(
      container.querySelector('.recharts-cartesian-axis.yAxis'),
    ).not.toBeInTheDocument();
  });

  test('both axes can be hidden', () => {
    const { container } = render(
      <HorzontalBarBlock
        {...props}
        axes={{
          ...axes,
          minor: {
            ...axes.minor,
            visible: false,
          },
          major: {
            ...axes.major,
            visible: false,
          },
        }}
      />,
    );

    expect(
      container.querySelector('.recharts-cartesian-axis.yAxis'),
    ).not.toBeInTheDocument();

    expect(
      container.querySelector('.recharts-cartesian-axis.xAxis'),
    ).not.toBeInTheDocument();
  });

  test('can hide legend', () => {
    const { container } = render(
      <HorzontalBarBlock {...props} legend="none" />,
    );

    expect(
      container.querySelector('.recharts-default-legend'),
    ).not.toBeInTheDocument();
  });
});
