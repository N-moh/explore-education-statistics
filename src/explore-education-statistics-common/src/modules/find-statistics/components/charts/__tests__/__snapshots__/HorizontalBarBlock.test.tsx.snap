// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorzontalBarBlock can stack data 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 900px; height: 300px;"
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 900px; height: 300px;"
    >
      <svg
        class="recharts-surface"
        height="300"
        version="1.1"
        viewBox="0 0 900 300"
        width="900"
      >
        <defs>
          <clippath
            id="recharts36-clip"
          >
            <rect
              height="270"
              width="810"
              x="90"
              y="0"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="0"
              y1="67.5"
              y2="67.5"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="0"
              y1="202.5"
              y2="202.5"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="0"
              y1="0"
              y2="0"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="0"
              y1="270"
              y2="270"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="110"
              x2="110"
              y="0"
              y1="0"
              y2="270"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="330"
              x2="330"
              y="0"
              y1="0"
              y2="270"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="550"
              x2="550"
              y="0"
              y1="0"
              y2="270"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="880"
              x2="880"
              y="0"
              y1="0"
              y2="270"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="90"
              y="0"
              y1="0"
              y2="270"
            />
            <line
              fill="none"
              height="270"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="900"
              x2="900"
              y="0"
              y1="0"
              y2="270"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            stroke="#666"
            width="810"
            x="90"
            x1="90"
            x2="900"
            y="270"
            y1="270"
            y2="270"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="110"
                x2="110"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="110"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="110"
                >
                  -3
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="330"
                x2="330"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="330"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="330"
                >
                  3
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="550"
                x2="550"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="550"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="550"
                >
                  9
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="880"
                x2="880"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="880"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="880"
                >
                  18
                </tspan>
              </text>
            </g>
          </g>
          <text
            class="recharts-text recharts-label"
            text-anchor="end"
            transform="rotate(-90, 90, 285)"
            x="90"
            y="285"
          >
            <tspan
              dy="0.355em"
              x="90"
            />
          </text>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="270"
            stroke="#666"
            width="60"
            x="30"
            x1="90"
            x2="90"
            y="0"
            y1="0"
            y2="270"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="270"
                stroke="#666"
                width="60"
                x="30"
                x1="84"
                x2="90"
                y="0"
                y1="67.5"
                y2="67.5"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="270"
                stroke="none"
                text-anchor="end"
                width="60"
                x="82"
                y="67.5"
              >
                <tspan
                  dy="0.355em"
                  x="82"
                >
                  2014/15
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="270"
                stroke="#666"
                width="60"
                x="30"
                x1="84"
                x2="90"
                y="0"
                y1="202.5"
                y2="202.5"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="270"
                stroke="none"
                text-anchor="end"
                width="60"
                x="82"
                y="202.5"
              >
                <tspan
                  dy="0.355em"
                  x="82"
                >
                  2015/16
                </tspan>
              </text>
            </g>
          </g>
          <text
            class="recharts-text recharts-label"
            text-anchor="middle"
            x="60"
            y="275"
          >
            <tspan
              dy="0.71em"
              x="60"
            />
          </text>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 220,13.5 h 110 v 108 h -110 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="110"
                x="220"
                y="13.5"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 220,148.5 h -110 v 108 h 110 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="-110"
                x="220"
                y="148.5"
              />
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 330,13.5 h 366.66666666666663 v 108 h -366.66666666666663 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="366.66666666666663"
                x="330"
                y="13.5"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 110,148.5 h 146.66666666666663 v 108 h -146.66666666666663 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="146.66666666666663"
                x="110"
                y="148.5"
              />
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 696.6666666666666,13.5 h 183.33333333333337 v 108 h -183.33333333333337 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="183.33333333333337"
                x="696.6666666666666"
                y="13.5"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 256.66666666666663,148.5 h 36.666666666666686 v 108 h -36.666666666666686 Z"
                fill="#ff0000"
                height="108"
                radius="0"
                stroke="#ff0000"
                width="36.666666666666686"
                x="256.66666666666663"
                y="148.5"
              />
            </g>
          </g>
        </g>
      </svg>
      <div
        class="recharts-tooltip-wrapper"
        style="pointer-events: none; visibility: hidden; position: absolute; top: 0px; transform: translate(undefinedpx, undefinedpx);"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</div>
`;

exports[`HorzontalBarBlock renders basic chart correctly 1`] = `
<div>
  <div
    class="recharts-responsive-container"
    style="width: 900px; height: 300px;"
  >
    <div
      class="recharts-wrapper"
      style="position: relative; cursor: default; width: 900px; height: 300px;"
    >
      <svg
        class="recharts-surface"
        height="300"
        version="1.1"
        viewBox="0 0 900 300"
        width="900"
      >
        <defs>
          <clippath
            id="recharts4-clip"
          >
            <rect
              height="260"
              width="810"
              x="90"
              y="10"
            />
          </clippath>
        </defs>
        <g
          class="recharts-cartesian-grid"
        >
          <g
            class="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="10"
              y1="75"
              y2="75"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="10"
              y1="205"
              y2="205"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="10"
              y1="10"
              y2="10"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="900"
              y="10"
              y1="270"
              y2="270"
            />
          </g>
          <g
            class="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="110"
              x2="110"
              y="10"
              y1="10"
              y2="270"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="346.9230769230769"
              x2="346.9230769230769"
              y="10"
              y1="10"
              y2="270"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="583.8461538461538"
              x2="583.8461538461538"
              y="10"
              y1="10"
              y2="270"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="880"
              x2="880"
              y="10"
              y1="10"
              y2="270"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="90"
              x2="90"
              y="10"
              y1="10"
              y2="270"
            />
            <line
              fill="none"
              height="260"
              stroke="#ccc"
              stroke-dasharray="3 3"
              width="810"
              x="90"
              x1="900"
              x2="900"
              y="10"
              y1="10"
              y2="270"
            />
          </g>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="30"
            stroke="#666"
            width="810"
            x="90"
            x1="90"
            x2="900"
            y="270"
            y1="270"
            y2="270"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="110"
                x2="110"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="110"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="110"
                >
                  -3
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="346.9230769230769"
                x2="346.9230769230769"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="346.9230769230769"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="346.9230769230769"
                >
                  1
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="583.8461538461538"
                x2="583.8461538461538"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="583.8461538461538"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="583.8461538461538"
                >
                  5
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="30"
                stroke="#666"
                width="810"
                x="90"
                x1="880"
                x2="880"
                y="270"
                y1="276"
                y2="270"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="30"
                stroke="none"
                text-anchor="middle"
                width="810"
                x="880"
                y="286"
              >
                <tspan
                  dy="0.71em"
                  x="880"
                >
                  10
                </tspan>
              </text>
            </g>
          </g>
          <text
            class="recharts-text recharts-label"
            text-anchor="end"
            transform="rotate(-90, 90, 285)"
            x="90"
            y="285"
          >
            <tspan
              dy="0.355em"
              x="90"
            />
          </text>
        </g>
        <g
          class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            class="recharts-cartesian-axis-line"
            fill="none"
            height="260"
            stroke="#666"
            width="60"
            x="30"
            x1="90"
            x2="90"
            y="10"
            y1="10"
            y2="270"
          />
          <g
            class="recharts-cartesian-axis-ticks"
          >
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                stroke="#666"
                width="60"
                x="30"
                x1="84"
                x2="90"
                y="10"
                y1="75"
                y2="75"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                stroke="none"
                text-anchor="end"
                width="60"
                x="82"
                y="75"
              >
                <tspan
                  dy="0.355em"
                  x="82"
                >
                  2014/15
                </tspan>
              </text>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                class="recharts-cartesian-axis-tick-line"
                fill="none"
                height="260"
                stroke="#666"
                width="60"
                x="30"
                x1="84"
                x2="90"
                y="10"
                y1="205"
                y2="205"
              />
              <text
                class="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height="260"
                stroke="none"
                text-anchor="end"
                width="60"
                x="82"
                y="205"
              >
                <tspan
                  dy="0.355em"
                  x="82"
                >
                  2015/16
                </tspan>
              </text>
            </g>
          </g>
          <text
            class="recharts-text recharts-label"
            text-anchor="middle"
            x="60"
            y="275"
          >
            <tspan
              dy="0.71em"
              x="60"
            />
          </text>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,23 h 177.69230769230768 v 32 h -177.69230769230768 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="177.69230769230768"
                x="287.69230769230774"
                y="23"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,153 h -177.69230769230774 v 32 h 177.69230769230774 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="-177.69230769230774"
                x="287.69230769230774"
                y="153"
              />
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,59 h 592.3076923076923 v 32 h -592.3076923076923 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="592.3076923076923"
                x="287.69230769230774"
                y="59"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,189 h 236.9230769230768 v 32 h -236.9230769230768 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="236.9230769230768"
                x="287.69230769230774"
                y="189"
              />
            </g>
          </g>
        </g>
        <g
          class="recharts-layer recharts-bar"
        >
          <g
            class="recharts-layer recharts-bar-rectangles"
          >
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,95 h 296.1538461538461 v 32 h -296.1538461538461 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="296.1538461538461"
                x="287.69230769230774"
                y="95"
              />
            </g>
            <g
              class="recharts-layer recharts-bar-rectangle"
            >
              <path
                class="recharts-rectangle"
                d="M 287.69230769230774,225 h 59.23076923076917 v 32 h -59.23076923076917 Z"
                fill="#ff0000"
                height="32"
                radius="0"
                stroke="#ff0000"
                width="59.23076923076917"
                x="287.69230769230774"
                y="225"
              />
            </g>
          </g>
        </g>
      </svg>
      <div
        class="recharts-legend-wrapper"
        style="position: absolute; width: 870px; height: 50px; left: 30px; top: 10px;"
      >
        <ul
          class="recharts-default-legend"
          style="padding: 0px; margin: 0px; text-align: center;"
        >
          <li
            class="recharts-legend-item legend-item-0"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#ff0000"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
            >
              Unauthorised absence rate
            </span>
          </li>
          <li
            class="recharts-legend-item legend-item-1"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#ff0000"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
            >
              Overall absence rate
            </span>
          </li>
          <li
            class="recharts-legend-item legend-item-2"
            style="display: inline-block; margin-right: 10px;"
          >
            <svg
              class="recharts-surface"
              height="14"
              style="display: inline-block; vertical-align: middle; margin-right: 4px;"
              version="1.1"
              viewBox="0 0 32 32"
              width="14"
            >
              <path
                class="recharts-legend-icon"
                d="M0,4h32v24h-32z"
                fill="#ff0000"
                stroke="none"
              />
            </svg>
            <span
              class="recharts-legend-item-text"
            >
              Authorised absence rate
            </span>
          </li>
        </ul>
      </div>
      <div
        class="recharts-tooltip-wrapper"
        style="pointer-events: none; visibility: hidden; position: absolute; top: 0px; transform: translate(undefinedpx, undefinedpx);"
      >
        <div
          class="recharts-default-tooltip"
          style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
        >
          <p
            class="recharts-tooltip-label"
            style="margin: 0px;"
          />
        </div>
      </div>
    </div>
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</div>
`;
